code/datums/antagonists/datum_traitor.dm

```diff
@@ -0,0 +1,241 @@
+			if(2)
+				var/datum/objective/purge/purge_objective = new
+				purge_objective.owner = owner
+				add_objective(purge_objective)
+			if(3)
+				var/datum/objective/robot_army/robot_objective = new
+				robot_objective.owner = owner
+				add_objective(robot_objective)
+			if(4) //Protect and strand a target
+				var/datum/objective/protect/yandere_one = new
+				yandere_one.owner = owner
+				add_objective(yandere_one)
+				yandere_one.find_target()
+				var/datum/objective/maroon/yandere_two = new
+				yandere_two.owner = owner
+				yandere_two.target = yandere_one.target
+				yandere_two.update_explanation_text() // normally called in find_target()
+				add_objective(yandere_two)
+				.=2
+	else
+		if(prob(50))
+			var/list/active_ais = active_ais()
+			if(active_ais.len && prob(100/GLOB.joined_player_list.len))
+				var/datum/objective/destroy/destroy_objective = new
+				destroy_objective.owner = owner
+				destroy_objective.find_target()
+				add_objective(destroy_objective)
+			else if(prob(30))
+				var/datum/objective/maroon/maroon_objective = new
+				maroon_objective.owner = owner
+				maroon_objective.find_target()
+				add_objective(maroon_objective)
+			else
+				var/datum/objective/assassinate/kill_objective = new
+				kill_objective.owner = owner
+				kill_objective.find_target()
+				add_objective(kill_objective)
+		else
+			var/datum/objective/steal/steal_objective = new
+			steal_objective.owner = owner
+			steal_objective.find_target()
+			add_objective(steal_objective)
+
+/datum/antagonist/traitor/greet()
+	to_chat(owner.current, "<B><font size=3 color=red>You are the [owner.special_role].</font></B>")
+	owner.announce_objectives()
+	if(should_give_codewords)
+		give_codewords()
+
+/datum/antagonist/traitor/proc/finalize_traitor()
+	if(issilicon(owner.current))
+		add_law_zero()
+		owner.current.playsound_local('sound/ambience/antag/Malf.ogg',100,0)
+		owner.current.grant_language(/datum/language/codespeak)
+	else
+		equip()
+		owner.current.playsound_local('sound/ambience/antag/TatorAlert.ogg',100,0)
+	SSticker.mode.update_traitor_icons_added(owner)
+	return
+
+/datum/antagonist/traitor/proc/give_codewords()
+	if(!owner.current)
+		return
+	var/mob/traitor_mob=owner.current
+
+	to_chat(traitor_mob, "<U><B>The Syndicate provided you with the following information on how to identify their agents:</B></U>")
+	to_chat(traitor_mob, "<B>Code Phrase</B>: <span class='danger'>[GLOB.syndicate_code_phrase]</span>")
+	to_chat(traitor_mob, "<B>Code Response</B>: <span class='danger'>[GLOB.syndicate_code_response]</span>")
+
+	traitor_mob.mind.store_memory("<b>Code Phrase</b>: [GLOB.syndicate_code_phrase]")
+	traitor_mob.mind.store_memory("<b>Code Response</b>: [GLOB.syndicate_code_response]")
+
+	to_chat(traitor_mob, "Use the code words in the order provided, during regular conversation, to identify other agents. Proceed with caution, however, as everyone is a potential foe.")
+
+/datum/antagonist/traitor/proc/add_law_zero()
```

Comment:
Maybe you're not willing to do this in this pr, but ideally the AI traitor datum would be a subtype of the parent traitor datum that only handles AI specific traitoring stuff.

That way the parent traitor datum is cleaner and the special sauce is confined to the child datum.

THen at runtime you can choose which traitor datum type to apply based on mob type.

Let me know if you could do that, or otherwise I'm happy to merge this.